index.html

```diff
@@ -10,184 +10,54 @@
-               ng-change="add(flatTaken)" ng-model-options="{debounce: 500}">
-      </td>
-      <td class="percent">
-        <input class="form-control input-sm" type="number" placeholder="+/-" ng-model="modifier.value"
-               ng-change="add(flatTaken)" ng-model-options="{debounce: 500}" id="flatTaken">
-      </td>
-      <td>
-        <button class="btn btn-danger btn-sm" ng-click="delete(flatTaken, $index)">Delete</button>
-      </td>
-    </tr>
-  </table>
-  <h5><b>Total Flat Reductions: {{flatTaken.total}}</b></h5>
-  <hr>
-  <table class="table table-condensed">
-    <tr ng-repeat="modifier in reducedTaken.values">
-      <td class="enabled">
-        <input type="checkbox" ng-model="modifier.enabled" title="">
-      </td>
-      <td class="name">
-        <input class="form-control input-sm" type="text" placeholder="Reduced Taken" ng-model="modifier.name"
-               ng-change="add(reducedTaken)" ng-model-options="{debounce: 500}">
-      </td>
-      <td class="percent">
-        <div class="input-group">
-          <input class="form-control input-sm" type="number" placeholder="+/-" ng-model="modifier.value"
-                 ng-change="add(reducedTaken)" ng-model-options="{debounce: 500}">
-          <div class="input-group-addon">%</div>
-        </div>
-      </td>
-      <td>
-        <button class="btn btn-danger btn-sm" ng-click="delete(reducedTaken, $index)">Delete</button>
-      </td>
-    </tr>
-  </table>
-  <h5><b>Total Reduced Damage Taken: {{reducedTaken.total}}%</b></h5>
-  <hr>
-  <table class="table table-condensed">
-    <tr ng-repeat="modifier in lessTaken.values">
-      <td class="enabled">
-        <input type="checkbox" ng-model="modifier.enabled" title="">
-      </td>
-      <td class="name">
-        <input class="form-control input-sm" type="text" placeholder="Less Taken" ng-model="modifier.name"
-               ng-change="add(lessTaken)" ng-model-options="{debounce: 500}">
-      </td>
-      <td class="percent">
-        <div class="input-group">
-          <input class="form-control input-sm" type="number" placeholder="+/-" ng-model="modifier.value"
-                 ng-change="add(lessTaken)" ng-model-options="{debounce: 500}">
-          <div class="input-group-addon">%</div>
-        </div>
-      </td>
-      <td>
-        <button class="btn btn-danger btn-sm" ng-click="delete(lessTaken, $index)">Delete</button>
-      </td>
-    </tr>
-  </table>
+    <table class="table table-condensed" ng-repeat-start="(tableName, table) in section.tables">
+      <tr ng-repeat="value in table.values">
+        <td>
+          <input type="checkbox" ng-model="value.enabled" ng-change="updateTotal(table)">
+        </td>
+        <td class="name">
+          <input class="form-control input-sm" type="text" placeholder={{table.placeholder}} ng-model="value.name"
+                 ng-change="add(table)" ng-model-options="{debounce: 500}">
+        </td>
+        <td class="percent">
```

Comment:
This is slightly problematic because flat damage reductions/increases are not a percent, so I'll need to break out this logic a bit